### PR TITLE
DOM-api done

### PR DESCRIPTION
**Pull request by mxmgny**
### DOM

[Demo](https://mxmgny.github.io/DOM_Practice/) | [Code base](https://github.com/mxmgny/frontend-2019-p2p/tree/dom-api-task/submissions/mxmgny/js-dom)

The code is submitted in its own sub-directory and in a dedicated feature branch.
Please, review.

I have also [opened PR](https://github.com/kottans/frontend-2019-homeworks/pull/384) to the frontend-2019-homeworks repo. What should I do with it?

### CODE REVIEW

File: submissions/mxmgny/script/script.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
+const requestURL = "https://raw.githubusercontent.com/mxmgny/DOM_Practice/master/script/data.json";
+const request = new XMLHttpRequest();
```
Comment (beta-version-profile) [inline submissions/mxmgny/script/script.js]: Try to rewrite the request for Fetch. Try to rewrite the request for Fetch. If you will need help please let me know
https://developer.mozilla.org/ru/docs/Web/API/Fetch_API/Using_Fetch

File: submissions/mxmgny/index.html [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <meta http-equiv="X-UA-Compatible" content="ie=edge">
+    <title>Warcraft races</title>
+    <link rel="stylesheet" type="text/css" href="style\style.css">
+    <script src="script/script.js" async></script>
+</head>
+<body>
+
+</body>
+</html>
```
Comment (o-msh) [inline submissions/mxmgny/index.html]: Please add empty line at the end of every files

File: submissions/mxmgny/script/script.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+function createTextElement(elementTag, text){
+    let element =  document.createElement(elementTag);
+    element.innerText = text;
+    return element;
+}
+
+function deactivate(tagName) { 
+    let arrayOfElements = Array.from(document.getElementsByTagName(tagName));
+    arrayOfElements.map((element) => {element.classList.value = ""});
```
Comment (o-msh) [inline submissions/mxmgny/script/script.js]: ```arrayOfElements.map(element => element.classList.value = "");``` will be better. Curly brackets aren’t required if only one expression is present.

File: submissions/mxmgny/style/style.css [hunk 4]
```diff
@@ -0,0 +1,70 @@
+aside{
+    width: 250px;
+    background-color:#f1f1f1;
+    position: absolute;
+    left: 0;
+    right: 0;
+    /* display: none; */
```
Comment (o-msh) [inline submissions/mxmgny/style/style.css]: Don't use unnecessary comments. Code will be more clean.

File: submissions/mxmgny/style/style.css [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    }
+button.active{
```
Comment (o-msh) [inline submissions/mxmgny/style/style.css]: Avoid styling by tag add custom styles on class.

File: submissions/mxmgny/script/script.js [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+function loadPage(jsonObj) {
+    const races = jsonObj['items'];
+
+    races.map( (race,id) => {
+
+        const navButton = createTextElement('button',race['header']);
+        navButton.addEventListener('click', changeRace);
+        if(id == 0) navButton.classList.add('active');
```
[the remark this answers is not in the document]
Comment (mxmgny) [reply submissions/mxmgny/script/script.js]: Thank you for your review. I updated my code so there's no comparisons.

File: submissions/mxmgny/script/script.js (hunk 1 above)
Comment (mxmgny) [reply submissions/mxmgny/script/script.js]: I refactor the request code using Fetch. Please review and comment if have more question and remarks.

File: submissions/mxmgny/script/script.js [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    loadPage(request.response);
+}
+
+function loadPage(jsonObj) {
+    const races = jsonObj['items'];
+
+    races.map( (race,id) => {
+
+        const navButton = createTextElement('button',race['header']);
+        navButton.addEventListener('click', changeRace);
```
[the remark this answers is not in the document]
Comment (mxmgny) [reply submissions/mxmgny/script/script.js]: I used event delegation in a new commit. Take a look, please, and point me to my mistakes if such exists.

File: submissions/mxmgny/index.html (hunk 2 above)
Comment (mxmgny) [reply submissions/mxmgny/index.html]: Thank you for your attention! I resolve this problem

File: submissions/mxmgny/script/script.js (hunk 3 above)
Comment (mxmgny) [reply submissions/mxmgny/script/script.js]: I make the code look better and replace Map method with forEach.  I think this method fits better in my function

**Comment by babayK0** [comment]
@mxmgny Please resolve outdated discussions

**Comment by babayK0** [comment]
@mxmgny Please resolve outdated discussions

File: submissions/mxmgny/style/style.css (hunk 5 above)
Comment (mxmgny) [reply submissions/mxmgny/style/style.css]: okay, thanks, I thought for this task it is acceptable.

Comment (o-msh) [reply submissions/mxmgny/style/style.css]: @mxmgny Styling by tag name is a bad practice in general. Also, you can read about [BEM](https://ru.bem.info/methodology/quick-start/). Using proper naming will prepare you for the changes in design of the website in future.

Comment (mxmgny) [reply submissions/mxmgny/style/style.css]: Thank you for that article it was very interesting and useful. I implemented BEM approach in my code. Please review if you can

**Comment by beta-version-profile** [comment]
Great job! Your fixes were really fast.

**Comment by o-msh** [comment]
@mxmgny You have improved the code noticeably. Well done!
But i have some comments, pay attention in the future.
 * `console.log` statements should not be left in final version of the code, unless it's part of the functionality. Users will not know what happened if you leave this instruction. IMO you can write just alert. But for other cases best practice will be write nice message to page.
* Always use `forEach` if you not necessary array transformation.
* Always put spaces around operators ```=``` ``` +``` ```-``` ```*``` ```/```, and after commas. This make your code more readable for humans. You can read about code style [here](https://github.com/airbnb/javascript)

**Pull request merged**